File: leetcode/1817.查找用户活跃分钟数.py
#
# @lc app=leetcode.cn id=1817 lang=python3
#
# [1817] 查找用户活跃分钟数
#
from typing import List
from collections import defaultdict


# @lc code=start
class Solution:
    def findingUsersActiveMinutes(self, logs: List[List[int]], k: int) -> List[int]:
        res = [0] * k
        cache_map = defaultdict(set)
        for user, minute in logs:
            cache_map[user].add(minute)
        for minute_set in cache_map.values():
            res[len(minute_set) - 1] += 1
        return res
# @lc code=end

File: leetcode/1817.查找用户活跃分钟数.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <unordered_set>
using namespace std;
/*
 * @lc app=leetcode.cn id=1817 lang=cpp
 *
 * [1817] 查找用户活跃分钟数
 */

// @lc code=start
class Solution
{
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>> &logs, int k)
    {
        vector<int> res(k, 0);
        unordered_map<int, unordered_set<int>> cache_map;
        for (auto &&log : logs)
        {
            cache_map[log[0]].emplace(log[1]);
        }
        for (auto &&i : cache_map)
        {
            res[i.second.size() - 1] += 1;
        }
        return res;
    }
};
// @lc code=end
